R/print.data.table.R

```diff
@@ -87,7 +89,19 @@ print.data.table = function(x, topn=getOption("datatable.print.topn"),
     toprint = rbind(abbs, toprint)
     rownames(toprint)[1L] = ""
   }
+  if (isFALSE(class) || (isTRUE(class) && col.names == "none")) abbs = ""
   if (quote) colnames(toprint) <- paste0('"', old <- colnames(toprint), '"')
+  if (isTRUE(trunc.cols)) {
+    # allow truncation of columns to print only what will fit in console PR #4074
+    widths = dt_width(toprint, class, row.names, col.names)
+    cons_width = getOption("width")
+    cols_to_print = widths <= cons_width
+    not_printed = colnames(toprint)[!cols_to_print]
+    if (sum(cols_to_print) == 0L) stop("Width of console too small to print a single column when `trunc.cols=TRUE`. Consider increasing the width of the console or use `trunc.cols=FALSE`.", call. = FALSE)
```

Comment:
this should be `!any(cols_to_print)`.

I also think we shouldn't `stop` here -- rather unfortunate for the program to interrupt from printing side effects. I would simply print nothing (besides the "variable not shown..." message).

Reply:
Nice. That makes a lot of sense to let it run with only the message itself.